R/backend-postgres.R

```diff
@@ -137,16 +137,26 @@ sql_translation.PqConnection <- function(con) {
         )
       },
       # https://www.postgresql.org/docs/current/functions-matching.html
-      str_like = function(string, pattern, ignore_case = TRUE) {
-        check_bool(ignore_case)
-        if (isTRUE(ignore_case)) {
-          sql_expr(!!string %ILIKE% !!pattern)
-        } else {
-          sql_expr(!!string %LIKE% !!pattern)
+      str_like = function(string, pattern, ignore_case = deprecated()) {
+
+        if (lifecycle::is_present(ignore_case)) {
+          lifecycle::deprecate_warn(
```

Comment:
Given that you've had to copy this three times, it would make sense to extract it into a function.

Reply:
Sure - happy to add that if you are happy with this https://github.com/tidyverse/dbplyr/pull/1644#discussion_r2583847343 (if you prefer that it just errors for all backends when ignore_case = TRUE then this code wouldn't be needed)